### PR TITLE
Fix Binary bounds

### LINKED ISSUE #171: Binary variables with bounds are incorrect
See https://discourse.julialang.org/t/xpress-julia-api-has-bug-with-fixing-variables/82895/4

```Julia
julia> model = Model(Xpress.Optimizer)
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: Xpress

julia> @variable(model, x, Bin)
x

julia> @objective(model, Max, x)
x

julia> fix(x, 0; force = true)

julia> optimize!(model)
FICO Xpress v8.13.3, Hyper, solve started 0:09:54, Jun 17, 2022
Heap usage: 74KB (peak 74KB, 360KB system)
Maximizing MILP  using up to 8 threads, with these control settings:
OUTPUTLOG = 1
MPSNAMELENGTH = 64
CALLBACKFROMMASTERTHREAD = 1
Original problem has:
         0 rows            1 cols            0 elements         1 globals
Presolved problem has:
         0 rows            1 cols            0 elements         1 globals
Presolve finished in 0 seconds
Heap usage: 75KB (peak 89KB, 362KB system)

Coefficient range                    original                 solved
  Coefficients   [min,max] : [      0.0,       0.0] / [      0.0,       0.0]
  RHS and bounds [min,max] : [ 1.00e+00,  1.00e+00] / [ 1.00e+00,  1.00e+00]
  Objective      [min,max] : [ 1.00e+00,  1.00e+00] / [ 1.00e+00,  1.00e+00]
Autoscaling applied standard scaling

Will try to keep branch and bound tree memory usage below 10.7GB
 *** Solution found:      .000000   Time:   0    Heuristic: T ***
 *** Solution found:     1.000000   Time:   0    Heuristic: T ***
STOPPING - MIPRELSTOP target reached (MIPRELSTOP=0.0001  gap=-0).
 *** Search completed ***
Uncrunching matrix
Final MIP objective                   : 1.000000000000000e+00
Final MIP bound                       : 1.000000000000000e+00
  Solution time / primaldual integral :         0s/ 92.741939%
  Number of solutions found / nodes   :         2 /         0
  Max primal violation      (abs/rel) :       0.0 /       0.0
  Max integer violation     (abs    ) :       0.0

julia> value(x)
1.0
```

There are tests for this in MOI.
https://github.com/jump-dev/MathOptInterface.jl/blob/b7600485d29d6cc72e5904d03a4fd47b50cac783/src/Test/test_variable.jl#L395-L419
Is it not being run for some reason?

### PR DESCRIPTION
**Pull request by rafabench**
Close #171
Also, this fix problems like:
```julia
model = Model(Xpress.Optimizer)
@variable(model, x, Bin)
@objective(model, Min, x)
fix(x,1, force=true)
unfix(x)
optimize!(model)
```
That would give unbounded.

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -1682,6 +1682,14 @@ function MOI.add_constraint(
         end
     end
     Xpress.chgcoltype(model.inner, [info.column], Cchar['B'])
+    if info.type == CONTINUOUS
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:1685]: Might be worth adding a comment above this, along the lines of `chgcoltype` resets variable bounds to `[0, 1]`, so we need to re-set them if they have been changed.

**Review by odow — APPROVED** [review]
Nice. I see the problem in our test in MOI. I'll update `MOI.Test` so that we catch things like this in the future.

**Comment by rafabench** [comment]
> Nice. I see the problem in our test in MOI. I'll update `MOI.Test` so that we catch things like this in the future.

Great! The delete constraint test is very important too.

File: test/MathOptInterface/MOI_wrapper.jl [hunk 2]
```diff
@@ -81,6 +81,45 @@ function test_runtests()
     return
 end
 
+function test_Binaryfixing()
+    @testset "Binary Equal to 1" begin
+        T = Float64
+        config = MOI.Test.Config(atol = 1e-3, rtol = 1e-3)
+        model = Xpress.Optimizer(OUTPUTLOG = 0)
+        x = MOI.add_variable(model)
+        MOI.add_constraint(model, x, MOI.EqualTo(T(1)))
```
Comment (joaquimg) [inline test/MathOptInterface/MOI_wrapper.jl:90]: what happens if we do:
`MOI.add_constraint(model, x, MOI.GreaterThan(T(1//2)))`

Comment (rafabench) [reply test/MathOptInterface/MOI_wrapper.jl:90]: It works, he gives the right bound
```
Minimize
 C1

Subject To
R1: C1 >= 0.5 

Bounds

Binaries
C1 

End
```

**Comment by rafabench** [comment]
The tests passed on MOI v1.6.0
image
image

**Pull request merged**